### PR TITLE
C2 amr a raouf c3 s26 google signin

### PR DESCRIPTION
**Pull request by AmrARaouf**
[#361]

@Omar-Hossam @Mirna-Yacout @karimngr @SalmaFG

### CODE REVIEW

File: arability/app/controllers/authentications_controller.rb [hunk 1]
```diff
@@ -141,15 +141,93 @@ def facebook_callback
...
+            flash[:success] = t(:signed_in_google)
+            sign_in_and_redirect(:gamer, auth.gamer)
+          else
+            flash[:error] = t(:no_account)
+            redirect_to "/gamers/sign_up"
+          end
+        else
+          gamer = Gamer.find_by_email(email)
+          if gamer
+            Authentication.create_with_omniauth(provider,gid,
```
Comment (omar-mohyeldin) [inline arability/app/controllers/authentications_controller.rb]: please insert space after comma

File: arability/app/controllers/authentications_controller.rb [hunk 2]
```diff
@@ -141,15 +141,93 @@ def facebook_callback
...
+            session["devise.token_secret"] = refresh_token
+            flash[:info] = t(:continue_reg_google)
+            redirect_to controller: "social_registrations",
+            action: "new_social", email: email, username: username,
+            gender: gender, provider: provider
+          end
+        end
+      else
+        if !auth
+          Authentication.create_with_omniauth(provider,gid,
```
Comment (omar-mohyeldin) [inline arability/app/controllers/authentications_controller.rb]: insert space after comma

File: arability/app/views/devise/registrations/new_social.html.erb [hunk 3]
```diff
@@ -15,22 +15,18 @@
...
-        <div class="control-group <%= @provider == 'twitter' ? ' ' : 'success' %>">
-            <div class="controls">
-                <%= f.label t(:email), class: "control-label" %>
-                <% if @provider == "twitter" %>
-                  <%= f.email_field :email, style: "margin-left: 10px;", value: @email %>
-                <% else %>
-                  <%= f.email_field :email, :style => "margin-left: 10px;", :value => @email, :readonly => "readonly", :id => "inputSuccess"  %>
-                <% end %>
-            </div>
+        <div class="control-group <%= (@provider == 'twitter') or (@errors and @errors.length > 0) ? ' ' : 'success' %>">
```
Comment (omar-mohyeldin) [inline arability/app/views/devise/registrations/new_social.html.erb]: please use double quotations

File: arability/app/views/devise/registrations/new_social.html.erb [hunk 4]
```diff
@@ -15,22 +15,18 @@
...
+            <%= f.label t(:email), class: "control-label" %>
+            <% if (@provider == 'twitter') or (@errors and @errors.length > 0) %>
+              <%= f.email_field :email, style: "margin-left: 10px;", value: @email %>
+            <% else %>
+                <%= f.email_field :email, :style => "margin-left: 10px;", :value => @email, :readonly => "readonly", :id => "inputSuccess"  %>
```
Comment (omar-mohyeldin) [inline arability/app/views/devise/registrations/new_social.html.erb]: avoid using =>

File: arability/app/views/devise/registrations/new_social.html.erb [hunk 5]
```diff
@@ -15,22 +15,18 @@
...
+            <% end %>
         </div></br>
 
         <div class="control-group success">
-            <div class="controls">
-                <%= f.label t(:username), :class => "control-label" %>
-                <%= f.text_field :username, :style => "margin-left: 10px;", :value => @username, :id => "inputSuccess" %>
-            </div>
+            <%= f.label t(:username), :class => "control-label" %>
+            <%= f.text_field :username, :style => "margin-left: 10px;", :value => @username, :id => "inputSuccess" %>
```
Comment (omar-mohyeldin) [inline arability/app/views/devise/registrations/new_social.html.erb]: avoid using => for the previous 2 lines

File: arability/app/views/devise/registrations/new_social.html.erb [hunk 6]
```diff
@@ -55,13 +51,13 @@
...
             <span style="margin-left: 10px;">
-                <%= f.country_select :country, [ "Egypt", "Lebanon", "Jordan" ]%>
+                <%= f.country_select :country, ["Egypt", "Lebanon", "Jordan" ]%>
             </span>
         <% end %></div></br>
 
         <div><%= f.label t(:education_level), :class => "control-label" %>
         <% if I18n.locale == :ar %>
-            <%= f.select(:education_level, options_for_select(['',['مدرسة','School'],['جامعة','University'],
+            <%= f.select(:education_level, options_for_select([['',''],['مدرسة','School'],['جامعة','University'],
```
Comment (omar-mohyeldin) [inline arability/app/views/devise/registrations/new_social.html.erb]: use double quotations

File: arability/config/initializers/omniauth.rb [hunk 7]
```diff
@@ -1,13 +1,20 @@
...
+  provider :twitter, "y3D1eGgnppUGjKNP6C47GQ", "37WC5Xa3HnawQe5A6YIMYSsTmUSzkKuky6gOpzD2I"
+
+  provider :facebook, "155521981280754", "a4d4684f0554e3637085e72eaa132890",
+    {scope: "email, offline_access, publish_stream",
+  	client_options: {ssl: {ca_path: "/etc/ssl/certs"}}}
+
+  provider :google_oauth2, ENV["GOOGLE_ID"], ENV["GOOGLE_SECRET"],
+           {
+             :scope => "userinfo.email,userinfo.profile,plus.me,http://gdata.youtube.com",
+             :approval_prompt => "auto"
```
Comment (omar-mohyeldin) [inline arability/config/initializers/omniauth.rb]: avoid using => for the previous 2 lines

**Comment by salmafg** [comment]
Documentation verified.

**Comment by omar-mohyeldin** [comment]
@AmrARaouf please modify your code

**Comment by salmafg** [comment]
Testing verified.

**Comment by AmrARaouf** [comment]
@Omar-Hossam please re review

**Comment by omar-mohyeldin** [comment]
@AmrARaouf code verified

File: arability/app/controllers/authentications_controller.rb [hunk 8]
```diff
@@ -141,15 +141,86 @@ def facebook_callback
...
+      gender = hash["extra"]["raw_info"]["gender"]
+      provider = hash["provider"]
+      gid = hash["uid"]
+      token = hash["credentials"]["token"]
+      refresh_token = hash["credentials"]["refresh_token"]
+      auth = Authentication.find_by_provider_and_gid(provider, gid)
+      if !gamer_signed_in?
+        if auth
+          if auth.gamer
+            flash[:success] = t(:signed_in_google)
```
Comment (Mirna-Yacout) [inline arability/app/controllers/authentications_controller.rb:194]: the if auth.gamer is not present in ur UML

**Comment by Mirna-Yacout** [comment]
- check comments, missing if block from UML

**Comment by AmrARaouf** [comment]
@Mirna-Yacout where are the comments?

**Comment by Mirna-Yacout** [comment]
in your code, check the last discussed diff

**Comment by ghost** [comment]
verified

**Comment by AmrARaouf** [comment]
@Mirna-Yacout please re review

**Comment by Mirna-Yacout** [comment]
verified

**Pull request merged**